rcl/src/rcl/wait.c

```diff
@@ -296,6 +296,7 @@ rcl_wait_set_get_allocator(const rcl_wait_set_t * wait_set, rcl_allocator_t * al
   } \
   rcl_allocator_t allocator = wait_set->impl->allocator; \
   wait_set->size_of_ ## Type ## s = 0; \
+  wait_set->impl->Type ## _index = 0; \
```

Comment:
Shouldn't this be set to the `min()` of the requested `size` and the `wait_set->impl->Type ## _index`? Otherwise this is also like a "clear" which is not the intention I think. If the size before was 4 and the index was 2, then the resize was to be 8, the index should still be 2 right?

Reply:
Look at the realloc code after that, it actually clear the new allocated memory. So, I think the old index is meaningless. If people want old index value, he should save it out before SET_RESIZE. 
  memset((void *)wait_set->Type ## s, 0, sizeof(rcl_ ## Type ## _t *) * size); \
  wait_set->size_of_ ## Type ## s = size; \

Reply:
You're right, it does zero the memory. :+1: